atox/src/main/kotlin/ui/user_profile/UserProfileFragment.kt

```diff
@@ -155,23 +170,101 @@ class UserProfileFragment : BaseFragment<FragmentUserProfileBinding>(FragmentUse
     }
 
     private fun createQrCodeDialog(): AlertDialog {
-        val qrData = QrCode.encodeText("tox:%s".format(vm.toxId.string()), QrCode.Ecc.LOW)
-        var bmp: Bitmap = Bitmap.createBitmap(qrData.size, qrData.size, Bitmap.Config.RGB_565)
-        for (x in 0 until qrData.size) {
-            for (y in 0 until qrData.size) {
-                bmp.setPixel(x, y, if (qrData.getModule(x, y)) Color.BLACK else Color.WHITE)
-            }
-        }
-        val metrics = resources.displayMetrics
-        val size = (min(metrics.widthPixels, metrics.heightPixels) * QR_CODE_TO_SCREEN_RATIO).roundToInt()
-        bmp = bmp.scale(size, size, false)
+        val bmp = createQrCode()
         val qrCode = ImageView(requireContext()).apply {
-            setPadding(dpToPx(QR_CODE_DIALOG_PADDING, resources))
+            setPadding(qrCodePadding.asPx(resources).px)
             setImageBitmap(bmp)
         }
+
         return AlertDialog.Builder(requireContext())
             .setTitle(R.string.tox_id)
             .setView(qrCode)
+            .setPositiveButton(getString(R.string.share)) { _, _ ->
+                vm.viewModelScope.launch {
+                    val qrImageUri = getQrForSharing("tox_id_qr_code")
+                    val shareIntent = Intent().apply {
+                        action = Intent.ACTION_SEND
+                        clipData = ClipData.newRawUri(null, qrImageUri)
+                        type = "image/png"
+                        putExtra(Intent.EXTRA_STREAM, qrImageUri)
+                        flags = Intent.FLAG_GRANT_READ_URI_PERMISSION
+                    }
+                    startActivity(Intent.createChooser(shareIntent, getString(R.string.tox_id_share)))
+                }
+            }
+            .setNegativeButton(getString(R.string.close), null)
```

Comment:
Maybe `Share` and `Cancel` as the text instead? You can imagine the QR view on this dialog as a share preview. Follows the convention for all other dialogs in aTox as well.

Reply:
But it's not really a process to cancel. It's just viewing the QR for nearby people to scan. And apart from that you can also share it.

Reply:
I'm going to remove all of this anyway with the redesign.

Reply:
Planning on removing it is all the more reason not to add a new string. Imo just drop the negative button in that case. The user can still close the dialog by hitting back or tapping outside of it.

Reply:
OK changing to `Cancel`.

Reply:
BTW. Why is there a `Cancel` string in aTox strings while this string is built-in in android strings like the `OK` string? Should I remove it? Or a seperate PR?

Reply:
Separate PR please! It's just an oversight that the not-built-in versions are being used. I think when I tried to do that last time I got a warning that the string was marked private in some dependency, so it didn't work out, but if that's no longer the case or I did something wrong, we should absolutely switch. :P

Reply:
OK but I'm still using the built-in for this specific one as it's a new code right?

Reply:
Either's fine! Personally I'd use the aTox one in this PR and switch all of them to the more correct way in a follow-up PR so it's consistent, but I don't mind either way!

Reply:
Yeah I will use the aTox one and then replace all of them in a PR.